core/src/main/java/org/elasticsearch/bootstrap/Security.java

```diff
@@ -212,7 +226,36 @@ static Permissions createPermissions(Environment environment) throws IOException
             // we just need permission to remove the file if its elsewhere.
             policy.add(new FilePermission(environment.pidFile().toString(), "delete"));
         }
-        return policy;
+    }
+    
+    static void addBindPermissions(Permissions policy, Settings settings) throws IOException {
+        // http is simple
+        String httpRange = settings.get("http.netty.port", 
+                               settings.get("http.port", 
+                                       NettyHttpServerTransport.DEFAULT_PORT_RANGE));
+        policy.add(new SocketPermission("localhost:" + httpRange, "listen,resolve"));
+        // transport is waaaay overengineered
+        Map<String, Settings> profiles = settings.getGroups("transport.profiles", true);
```

Comment:
> However, this strikes me as very fragile and subtle and I would want a minimum a test to ensure this behavior is preserved.

That is why instead I would just prefer to keep the security code contained and not "reach out" to other code. Even if its at the expense of a few constants or helper methods. Thats just my opinion.

We know we make this tradeoff by allowing things like paths and ports to be configurable in elasticsearch.yml, the other option is to remove that flexibility, then we could initialize SM on JVM startup?